### PR TITLE
AQI-3806 Added ClientSDK.ThrowAPIErrors everywhere and GetIntTwinDataProperty() not returning Int

### PR DESCRIPTION
**Pull request by DarshanaGandhi-AI**

### CODE REVIEW

File: src/Aquarius.ONE.ClientSDK/ClientSDK.cs [hunk 1]
```diff
@@ -91,52 +91,52 @@ public PlatformEnvironment Environment
...
 
             UserHelper = new UserHelper(Authentication, Core);
             CacheHelper = new CacheHelper(this);
 
-            Configuration = new ConfigurationApi(Environment, ContinueOnCapturedContext, _restHelper);
+            Configuration = new ConfigurationApi(Environment, ContinueOnCapturedContext, _restHelper, ThrowAPIErrors);
             Configuration.Event += Logger.Logger_Event;
 
             // This is a wrapper around the ConfigurationApi no events are logged directly
             Logbook = new LogbookApi(Environment, ContinueOnCapturedContext, _restHelper);
```
Comment (AQI-DanG) [inline src/Aquarius.ONE.ClientSDK/ClientSDK.cs:110]: LogbookApi creates a ConfigurationApi class so it needs to be able to pass ThrowApiErrors to that class.

File: src/Aquarius.ONE.ClientSDK/ClientSDK.cs [hunk 2]
```diff
@@ -91,52 +91,52 @@ public PlatformEnvironment Environment
...
 
-            Spreadsheet = new SpreadsheetApi(Environment, ContinueOnCapturedContext, _restHelper);
+            Spreadsheet = new SpreadsheetApi(Environment, ContinueOnCapturedContext, _restHelper, ThrowAPIErrors);
             Spreadsheet.Event += Logger.Logger_Event;
 
-            Data = new DataApi(Environment, ContinueOnCapturedContext, _restHelper);
+            Data = new DataApi(Environment, ContinueOnCapturedContext, _restHelper, ThrowAPIErrors);
             Data.Event += Logger.Logger_Event;
 
             Sample = new SampleApi(Environment, ContinueOnCapturedContext, _restHelper, this);
```
Comment (AQI-DanG) [inline src/Aquarius.ONE.ClientSDK/ClientSDK.cs:136]: For consistency I would suggest using the same pattern on all the api classes, either update this one to just take a bool or update the others to take a clientSdk if necessary.

**Review by AQI-DanG — CHANGES_REQUESTED** [review]
Couple small things that should be changed but I looks pretty good overall.

File: src/Aquarius.ONE.ClientSDK/Common/Activity/ActivityApi.cs [hunk 3]
```diff
@@ -83,7 +84,9 @@ public ActivityApi(PlatformEnvironment environment, bool continueOnCapturedConte
             catch (Exception e)
             {
                 Event(e, CreateLoggerArgs(EnumEventLevel.Error, $"GetActivitiesAsync Failed - {e.Message}"));
-                throw;
+                if (_throwAPIErrors) 
+					 throw;
```
Comment (JonathanRobertson-AQI) [inline src/Aquarius.ONE.ClientSDK/Common/Activity/ActivityApi.cs:88]: I think that the spacing / tabs might be inconsistent. When I open a file in Visual Studio, I get the following message. Can you please fix this in the files you updated?

image

File: src/Aquarius.ONE.ClientSDK/Common/Schedule/ScheduleApi.cs [hunk 4]
```diff
@@ -12,15 +11,17 @@ namespace ONE.Common.Schedule
...
-        public ScheduleApi(PlatformEnvironment environment, bool continueOnCapturedContext, RestHelper restHelper)
+        public ScheduleApi(PlatformEnvironment environment, bool continueOnCapturedContext, RestHelper restHelper, bool throwAPIErrors = false)
         {
             _environment = environment;
             _continueOnCapturedContext = continueOnCapturedContext;
             _restHelper = restHelper;
+            _throwAPIErrors = throwAPIErrors;
         }
         private PlatformEnvironment _environment;
         private bool _continueOnCapturedContext;
```
Comment (JonathanRobertson-AQI) [inline src/Aquarius.ONE.ClientSDK/Common/Schedule/ScheduleApi.cs:22]: It looks like a couple methods were missed - like Get Activitie

Comment (DarshanaGandhi-AI) [reply src/Aquarius.ONE.ClientSDK/Common/Schedule/ScheduleApi.cs:22]: GetOneActivity() method is done

**Review by KentMiller — APPROVED** [review]

**Review by AQI-DanG — APPROVED** [review]

**Comment by JonathanRobertson-AQI** [comment]
I'm still seeing the message about inconsistent line endings, please fix this... 

image

**Comment by JonathanRobertson-AQI** [comment]
It looks like a couple methods on SampleApi.cs are missing the check before throwing... 

image

**Review by JonathanRobertson-AQI — APPROVED** [review]

**Comment by JonathanRobertson-AQI** [comment]
Looks good to me now, thanks!

I'd advocate keeping this PR open, creating the pre-release package from your feature branch, and assigning it to @dedwards-AQI for testing.  When he approves test, then we could merge.

**Pull request merged**